File: python/calculate_mode.py
def calculate_mode(inp):
    #dict to use as map for each element to log value, count, and type..just because...
    ele_map = {}
    #answer list
    ans_list = []
    #looping through input list to populate the map
    for ele in inp:
        if ele in ele_map:
            ele_map[ele]['count'] += 1
        else :
            ele_map[ele] = {
                'value': ele,
                'count': 1,
            }
    #list to push item counts to
    item_count = []
    #looping through the map pushing counts to list
    for key in ele_map:
        item_count.append(ele_map[key]['count'])
    #using max function against item count to get max count number
    target = max(item_count)
    #looping through the map and appending value to answer list for any that match count value
    for key in ele_map:
        if ele_map[key]['count'] == target:
            ans_list.append(ele_map[key]['value'])
    #returning the answer list
    return ans_list

File: js/calculateMode.js
const calculateMode = (arr) => {
    //object to map value, count, and type of each index in input array
    const eleMap = {};
    //answer array to push final values into and return
    const ansArray = [];
    //looping through input array to populate the map
    for (let i = 0; i < arr.length; i++){
        //if the value is already present in map then increasing it's count value by 1
        if (eleMap[arr[i]]) {
            eleMap[arr[i]]['count'] += 1;
        //if it is not in map I am creating a key of the value and it's value is an object
        //with keys value, count, and type of the orginal value...just in case
        } else {
            eleMap[arr[i]] = {
                value: arr[i],
                count: 1,
                type: typeof(arr[i])
            }
        }
    }
    //array to push item counts to in order to get max count value
    const itemCount = []
    //looping through map and pushing counts to item count array
    for (let key in eleMap) {
        itemCount.push(eleMap[key].count)
    }
    //using Math.max to get target max count number
    const targetNum = Math.max(...itemCount)
    //looping through map again finding keys whose count match target number and pushing their 
    //values into the answer array
    for (let key in eleMap) {
        if (eleMap[key].count === targetNum) {
            ansArray.push(eleMap[key].value)
        }
    }
    //returning sorted array
    return ansArray.sort()
}
